profilingdemo/source/java/src/org/kaaproject/kaa/demo/profiling/ProfilingDemo.java

```diff
@@ -0,0 +1,100 @@
+/**
+ *  Copyright 2014-2016 CyberVision, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.kaaproject.kaa.demo.profiling;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+
+/**
+ * A demo application class that use the Kaa endpoint profiling and grouping API.
+ *
+ * @author Maksym Liashenko
+ */
+public class ProfilingDemo {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ProfilingDemo.class);
+
+    public static void main(String[] args) throws IOException {
+        LOG.info("Configuration demo started");
+
+        // we need folder to save properties in it. In other case we can't create more then one KaaClient
+        if (!Files.exists(Paths.get(KaaManager.PROPERTIES_OUT_DIR))) {
+            LOG.error("No resource folder. Configuration demo cancelled");
```

Comment:
This folder created by me and must be in res/ folder. In this folder there are public/private keys